File: CS/PS/Site/leetcode/hashmap/347.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function (nums, k) {
  let obj = {}
  let answer = []
  for (let i = 0; i < nums.length; i++) {
    if (!(nums[i] in obj)) {
      obj[nums[i]] = 1
    }
    else {
      obj[nums[i]] += 1;
    }
  }
  let arr = Object.entries(obj)
  arr = arr.sort((a, b) => b[1] - a[1]);
  for (let i = 0; i < k; i++) {
    answer.push(parseInt(arr[i][0]))
  }
  return answer
};

console.log(topKFrequent([1, 1, 1, 1, 2, 2, 2, 3, 3, 3, 3, 3, 3], 2));
console.log(topKFrequent([1, 2], 2));

File: CS/PS/Site/leetcode/hashmap/347.py
from typing import List


# class Solution:
#     def topKFrequent(self, nums: List[int], k: int) -> List[int]:
#         num_set = {}
#         answer = []
#         for v in nums:
#             if v not in num_set:
#                 num_set[v] = 1
#             else:
#                 num_set[v] += 1
#         num_list = list(num_set.items())
#         sorted_list = sorted(num_list, key=lambda x: (x[1]), reverse=True)
#         for idx in range(k):
#           answer.append(sorted_list[idx][0])
#         return answer
#     print(topKFrequent("", [1, 1, 1, 2, 2, 3], 2))
#     print(topKFrequent("", [1, 2], 2))

from collections import Counter


class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter_obj = Counter(nums).most_common(k)
        answer = []
        for v in counter_obj:
            answer.append(v[0])
        return answer
    print(topKFrequent("", [1, 1, 1, 2, 2, 3], 2))
